extensions/wikia/Forum/ForumHelper.class.php

```diff
@@ -2,15 +2,39 @@
 class ForumHelper {
 	public static $forumNamespaces = [NS_WIKIA_FORUM_BOARD, NS_WIKIA_FORUM_TOPIC_BOARD, NS_WIKIA_FORUM_BOARD_THREAD];
 
+	private static $isForumCached = null;
+
+	/**
+	 * @return bool
+	 */
 	public static function isForum() {
 		global $wgTitle;
-		if ( $wgTitle instanceof Title ) {
-			if ( $wgTitle->isSpecial( 'Forum' ) || in_array( $wgTitle->getNamespace(), self::$forumNamespaces ) ) {
+
+		// called to early?
+		if ( ( $wgTitle instanceof Title ) === false ) {
+			return false;
+		}
+
+		// this method can be called 30+ times on each page, cache the result as $wgTitle is not likely to change
+		if ( self::$isForumCached === null ) {
+			self::$isForumCached = self::isTitleForum( $wgTitle );
+		}
+
+		return self::$isForumCached;
+	}
+
+	/**
+	 * @param Title $title
+	 * @return bool
+	 */
+	private static function isTitleForum( $title ) {
+		if ( $title instanceof Title ) {
+			if ( $title->isSpecial( 'Forum' ) || in_array( $title->getNamespace(), self::$forumNamespaces ) ) {
 				return true;
-			} else if ( $wgTitle->getNamespace() === NS_USER_WALL_MESSAGE ) {
-				$mainTitle = Title::newFromId( $wgTitle->getText() );
+			} else if ( $title->getNamespace() === NS_USER_WALL_MESSAGE ) {
+				$mainTitle = Title::newFromID( $title->getText() );
```

Comment:
![It's by design](http://starecat.com/content/wp-content/uploads/bug-feature-by-design.jpg)